File: Service/Implement/CategoryStatusService.cs
namespace Service.Implement
{
    public class CategoryStatusService : BaseService<CategoryStatus, ICategoryStatusRepository>
    , ICategoryStatusService
    {
        private readonly ICategoryStatusRepository _CategoryStatusRepository;
        public CategoryStatusService(ICategoryStatusRepository CategoryStatusRepository) : base(CategoryStatusRepository)
        {
            _CategoryStatusRepository = CategoryStatusRepository;
        }
    }
}

File: ERP_Angular/src/app/shared/ERP/CategoryStatus.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { MatTableDataSource } from '@angular/material/table';
import { environment } from 'src/environments/environment';
import { CategoryStatus } from './CategoryStatus.model';
import { BaseService } from './Base.service';
import { BaseResult } from 'src/app/shared/ERP/BaseResult.model';

@Injectable({
    providedIn: 'root'
})
export class CategoryStatusService extends BaseService{
 DisplayColumns001: string[] = ['No', 'ID',  'Code', 'Name', 'Display', 'Note', 'SortOrder', 'Active', 'Save'];

    List: CategoryStatus[] | undefined;
    ListFilter: CategoryStatus[] | undefined;
    FormData!: CategoryStatus;
    constructor(public httpClient: HttpClient) {
        super(httpClient);
        this.Controller = "CategoryStatus";
    }
}

